services/brig/src/Brig/IO/Intra.hs

```diff
@@ -301,23 +348,37 @@ notifyUserDeletionRemotes ::
   UserId ->
   m ()
 notifyUserDeletionRemotes deleted = do
+  luid <- qualifyLocal deleted
   runConduit $
-    Data.lookupRemoteConnectedUsersC deleted (fromInteger (natVal (Proxy @UserDeletedNotificationMaxConnections)))
+    Data.lookupRemoteConnectedUsersC luid (fromInteger (natVal (Proxy @UserDeletedNotificationMaxConnections)))
       .| C.mapM_ fanoutNotifications
   where
-    fanoutNotifications :: [Remote UserId] -> m ()
+    fanoutNotifications :: [Remote UserConnection] -> m ()
     fanoutNotifications = mapM_ notifyBackend . bucketRemote
 
-    notifyBackend :: Remote [UserId] -> m ()
-    notifyBackend uids = do
-      case tUnqualified (checked <$> uids) of
+    notifyBackend :: Remote [UserConnection] -> m ()
+    notifyBackend ucs = do
+      case tUnqualified (checked <$> ucs) of
         Nothing ->
           -- The user IDs cannot be more than 1000, so we can assume the range
           -- check will only fail because there are 0 User Ids.
           pure ()
-        Just rangedUids -> do
+        Just rangedUcs -> do
           luidDeleted <- qualifyLocal deleted
-          notifyUserDeleted luidDeleted (qualifyAs uids rangedUids)
+          notifyUserDeleted luidDeleted (qualifyAs ucs ((fmap (fmap (qUnqualified . ucTo))) rangedUcs))
+          -- also sent connection cancelled events to the connections that are pending
+          let remotePendingConnections = qualifyAs ucs <$> filter ((==) Sent . ucStatus) (fromRange rangedUcs)
+          forM_ remotePendingConnections $ sendCancelledEvent luidDeleted
+
+    sendCancelledEvent :: Local UserId -> Remote UserConnection -> m ()
+    sendCancelledEvent luidDeleted ruc = do
+      runExceptT (sendConnectionAction luidDeleted Nothing (qUnqualified . ucTo <$> ruc) RemoteRescind) >>= \case
+        -- I don't think we want to abort the operation since this is running asynchronously and cannot be retried anyway
```

Comment:
why can't it be retried?  i'd expect this to go back to the rabbit queue and then get picked up again in a bit?